### PR TITLE
Migrate remaining Garden formulae to gz

### PR DESCRIPTION
**Pull request by chapulina**
* Part of https://github.com/gazebo-tooling/release-tools/issues/718

### CODE REVIEW

**Review by methylDragon — APPROVED** [review]

File: Formula/ignition-garden.rb [hunk 1]
```diff
@@ -47,7 +47,7 @@ def install
   end
 
   test do
-    yaml_file = share/"ignition/ignition-garden/gazebodistro/collection-garden.yaml"
+    yaml_file = share/"gz/gz-garden/gazebodistro/collection-garden.yaml"
```
Comment (scpeters) [inline Formula/ignition-garden.rb:50]: I think these install paths are not updated yet

* [Build Status](https://build.osrfoundation.org/job/ignition_garden-install_bottle-homebrew-amd64/233/) https://build.osrfoundation.org/job/ignition_garden-install_bottle-homebrew-amd64/233/

~~~
vcs validate: error: argument --input: can't open
 '/usr/local/Cellar/ignition-garden/0.999.999~0~20220414/share/gz/gz-garden/gazebodistro/collection-garden.yaml':
 [Errno 2] No such file or directory:
 '/usr/local/Cellar/ignition-garden/0.999.999~0~20220414/share/gz/gz-garden/gazebodistro/collection-garden.yaml'
~~~

Comment (chapulina) [reply Formula/ignition-garden.rb:50]: Yeah we haven't migrated `IGN_DATA_INSTALL_DIR` yet. This was future-proofing. I understood no bottles were being built yet.

Comment (scpeters) [reply Formula/ignition-garden.rb:50]: the garden formula is a special case; we have a nightly job to build that formula and run its test. I had forgotten about this when saying that tests were only used when building bottles. I'll submit a small PR to fix this formula only

Comment (scpeters) [reply Formula/ignition-garden.rb:50]: https://github.com/osrf/homebrew-simulation/pull/1948

**Pull request merged**